core/store/fs_directory.cpp

```diff
@@ -399,6 +416,27 @@ class pooled_fs_index_input final : public fs_index_input {
   file_handle::ptr reopen(const file_handle& src) const;
 };
 
+fs_index_input::fs_index_input(file_handle::ptr&& handle,
+                               size_t pool_size) noexcept
+  : handle_(std::move(handle)), pool_size_(pool_size), pos_(0) {
+  IRS_ASSERT(handle_);
+  handle_->resource_manager.readers->Increase(sizeof(pooled_fs_index_input));
+  buffered_index_input::reset(buf_, sizeof buf_, 0);
+}
+
+fs_index_input::fs_index_input(const fs_index_input& rhs) noexcept
+  : handle_(rhs.handle_), pool_size_(rhs.pool_size_), pos_(rhs.file_pointer()) {
+  IRS_ASSERT(handle_);
+  handle_->resource_manager.readers->Increase(sizeof(pooled_fs_index_input));
+  buffered_index_input::reset(buf_, sizeof buf_, pos_);
+}
+
+fs_index_input ::~fs_index_input() {
```

Comment:
I'm not sure clang-format fix this or not
```suggestion
fs_index_input::~fs_index_input() {
```